code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_darkrevive.dm

```diff
@@ -0,0 +1,75 @@
+/datum/artifact_effect/darkrevive
+	effecttype = "darkrevive"
+
+/datum/artifact_effect/darkrevive/New()
+	..()
+	effect = pick(EFFECT_TOUCH)
+	effect_type = pick(0,2,5)
+
+/datum/artifact_effect/darkrevive/DoEffectTouch(var/mob/living/carbon/human/user)
+	if(holder && !isplasmaman(user) && !isskellington(user) && !isgolem(user) && !user.isDead() && user.dna.mutantrace != "shadow")
+		var/list/targets = list()
+		FOR_DVIEW(var/mob/living/carbon/human/H,world.view,get_turf(holder),0)
+			if(!H.mind)
+				continue
+			if(isplasmaman(H) || isskellington(H) || isgolem(H))
```

Comment:
They check the type of a species - this is not the same as a species have its own code, like a proc, for whether or not it can do what you want i.e. `/datum/species/proc/is_revivable()`